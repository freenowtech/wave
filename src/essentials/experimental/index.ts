export * from './theme';
export { createGlobalStyle, ColorPalette } from './Colors';
export { darkTheme } from './globalStyles';
export { getSemanticValue, getBareValue, getBareHslValue, getSemanticHslValue } from './cssVariables';
