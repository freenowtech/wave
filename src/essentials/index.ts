export { Elevation } from './Elevation/Elevation';
export { Spaces } from './Spaces/Spaces';
export { Breakpoints, MediaQueries } from './Breakpoints/Breakpoints';
export type { ReadCssColorVariable, ReadCssVariable } from './Colors/types';
export { GlobalStyle as ClassicColors } from './Colors/Colors';
export { GlobalStyle as ModernColors } from './Colors/ModernColors';
