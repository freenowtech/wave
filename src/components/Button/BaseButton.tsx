import { ComponentPropsWithoutRef } from 'react';
import styled from 'styled-components';
import { compose, margin, MarginProps, ResponsiveValue, variant, width, WidthProps } from 'styled-system';

import { theme } from '../../essentials/theme';
import { deprecatedProperty } from '../../utils/deprecatedProperty';
import { get } from '../../utils/themeGet';

interface BaseButtonProps extends ComponentPropsWithoutRef<'button'>, MarginProps, WidthProps {
    /**
     * Adjusts the size of the button
     */
    size?: ResponsiveValue<'small' | 'medium'>;
    /**
     * Sets the button width to 100% of the parent
     * @deprecated
     */
    block?: boolean;
}

const sizeVariant = variant({
    prop: 'size',
    variants: {
        small: {
            fontSize: get('fontSizes.1'),
            height: '2rem',
            minWidth: '5rem',
            padding: '0 0.75rem'
        },
        medium: {
            fontSize: get('fontSizes.2'),
            height: '3rem',
            minWidth: '6.25rem',
            padding: '0 1rem'
        }
    }
});

/**
 * @deprecated
 */
function handleBlockProp({ block }: BaseButtonProps): string {
    if (block !== undefined) {
        deprecatedProperty('Button', block, 'block', 'width');
    }

    if (block) {
        return 'width: 100%';
    }

    return '';
}

// "svg path" fill set to "inherit" to being able to transition using button variants
const BaseButton = styled.button.attrs({ theme })<BaseButtonProps>`
    align-items: center;
    background: transparent;
    border-radius: ${get('radii.2')};
    border: 0.0625rem solid transparent;
    cursor: pointer;
    display: inline-flex;
    font-family: ${get('fonts.normal')};
    font-weight: ${get('fontWeights.semibold')};
    justify-content: center;
    text-align: center;
    text-decoration: none;
    ${handleBlockProp};

    &:disabled {
        cursor: not-allowed;
    }

    svg path {
        fill: inherit;
    }

    ${compose(margin, sizeVariant, width)}
`;

export { BaseButton, BaseButtonProps };
