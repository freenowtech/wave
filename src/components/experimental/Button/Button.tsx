import React, { ReactElement } from 'react';
import styled from 'styled-components';
import { variant } from 'styled-system';
import { Button as BaseButton, ButtonProps as BaseButtonProps } from 'react-aria-components';
import { getSemanticValue } from '../../../essentials/experimental/cssVariables';
import { get } from '../../../utils/experimental/themeGet';
import { textStyles } from '../Text/Text';

type Emphasis = 'primary' | 'secondary';

interface ButtonProps extends BaseButtonProps {
    /**
     * Define style of the button component, defaults to primary
     */
    emphasis?: Emphasis;
}

const emphasisStyles = variant<Record<string, unknown>, Emphasis>({
    prop: '$emphasis',
    variants: {
        primary: {
            color: getSemanticValue('on-accent'),
            background: getSemanticValue('accent'),

            '&::before': {
                background: getSemanticValue('on-accent')
            },

            '&[data-disabled]::before': {
                opacity: 0.38
            }
        },
        secondary: {
            color: getSemanticValue('on-surface'),
            background: getSemanticValue('surface-variant'),

            '&::before': {
                background: getSemanticValue('on-surface')
            },

            '&[data-disabled]': {
                opacity: 0.38
            },

            '&[data-disabled]::before': {
                opacity: 0.06
            }
        }
    }
});

const ButtonStyled = styled(BaseButton)<{ $emphasis: Emphasis }>`
    position: relative;

    display: inline-flex;
    align-items: center;
    gap: ${get('space.2')};
    border: none;
    outline: none;
    border-radius: ${get('radii.4')};
    padding: ${get('space.4')} ${get('space.6')};

    cursor: pointer;

    &[data-disabled] {
        cursor: not-allowed;
    }

    &::before {
        position: absolute;
        top: 0;
        right: 0;
        left: 0;
        bottom: 0;
        content: '';
        border-radius: inherit;
        opacity: 0;
        transition: opacity ease 200ms;
    }

    &[data-hovered]::before {
        opacity: 0.16;
    }

    &[data-focused]::before,
    &[data-pressed]::before {
        opacity: 0.24;
    }

    ${textStyles.variants.label1}

    ${emphasisStyles};
`;

function Button({ children, emphasis = 'primary', ...restProps }: ButtonProps): ReactElement {
    return (
        <ButtonStyled $emphasis={emphasis} {...restProps}>
            {children}
        </ButtonStyled>
    );
}

export { Button };
