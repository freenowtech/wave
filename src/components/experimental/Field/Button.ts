import styled from 'styled-components';
import { Button as BaseButton } from 'react-aria-components';
import { getSemanticValue } from '../../../essentials/experimental/cssVariables';
import { get } from '../../../utils/experimental/themeGet';

export const Button = styled(BaseButton)`
    appearance: none;
    background: none;
    border: none;
    display: flex;
    margin: 0;
    padding: 0;
    cursor: pointer;
    outline: 0;

    &[data-focused] {
        outline: ${getSemanticValue('interactive')} solid 0.125rem;
        border-radius: ${get('radii.2')};
    }
`;
