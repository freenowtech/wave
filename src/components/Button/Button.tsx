import * as React from 'react';
import styled from 'styled-components';
import { ButtonStyleProps, ResponsiveValue, variant } from 'styled-system';
import { theme } from '../../essentials/theme';
import { get } from '../../utils/themeGet';

import { BaseButton, BaseButtonProps } from './BaseButton';

interface ButtonProps extends BaseButtonProps, ButtonStyleProps {
    /**
     * Adjust colors for display on a dark background
     */
    inverted?: boolean;
    /**
     * Define style of the button component, defaults to primary
     */
    variant?: ResponsiveValue<'primary' | 'secondary' | 'danger'>;
}

const variantStyles = variant({
    variants: {
        primary: {
            color: get('semanticColors.button.primary.text'),
            fill: get('semanticColors.button.primary.icon'),
            background: get('semanticColors.button.primary.background'),
            borderColor: get('semanticColors.button.primary.border'),

            '&:hover': {
                color: get('semanticColors.button.primary.textHover'),
                fill: get('semanticColors.button.primary.iconHover'),
                background: get('semanticColors.button.primary.backgroundHover'),
                borderColor: get('semanticColors.button.primary.borderHover')
            },

            '&:disabled': {
                color: get('semanticColors.button.primary.textDisabled'),
                fill: get('semanticColors.button.primary.iconDisabled'),
                background: get('semanticColors.button.primary.backgroundDisabled'),
                borderColor: get('semanticColors.button.primary.borderDisabled')
            }
        },
        secondary: {
            color: get('semanticColors.button.secondary.text'),
            fill: get('semanticColors.button.secondary.icon'),
            background: get('semanticColors.button.secondary.background'),
            borderColor: get('semanticColors.button.secondary.border'),
            '&:hover': {
                color: get('semanticColors.button.secondary.textHover'),
                fill: get('semanticColors.button.secondary.iconHover'),
                background: get('semanticColors.button.secondary.backgroundHover'),
                borderColor: get('semanticColors.button.secondary.borderHover')
            },
            '&:disabled': {
                color: get('semanticColors.button.secondary.textDisabled'),
                fill: get('semanticColors.button.secondary.iconDisabled'),
                background: get('semanticColors.button.secondary.backgroundDisabled'),
                borderColor: get('semanticColors.button.secondary.borderDisabled')
            }
        },
        danger: {
            color: get('semanticColors.button.danger.text'),
            fill: get('semanticColors.button.danger.icon'),
            background: get('semanticColors.button.danger.background'),
            borderColor: get('semanticColors.button.danger.border'),

            '&:hover': {
                color: get('semanticColors.button.danger.textHover'),
                fill: get('semanticColors.button.danger.iconHover'),
                background: get('semanticColors.button.danger.backgroundHover'),
                borderColor: get('semanticColors.button.danger.borderHover')
            },

            '&:disabled': {
                color: get('semanticColors.button.danger.textDisabled'),
                fill: get('semanticColors.button.danger.iconDisabled'),
                background: get('semanticColors.button.danger.backgroundDisabled'),
                borderColor: get('semanticColors.button.danger.borderDisabled')
            }
        }
    }
});

const invertedVariantStyles = variant({
    variants: {
        primary: {
            color: get('semanticColors.button.primary.textInverted'),
            fill: get('semanticColors.button.primary.iconInverted'),
            background: get('semanticColors.button.primary.backgroundInverted'),
            borderColor: get('semanticColors.button.primary.borderInverted'),

            '&:hover': {
                color: get('semanticColors.button.primary.textHoverInverted'),
                fill: get('semanticColors.button.primary.iconHoverInverted'),
                background: get('semanticColors.button.primary.backgroundHoverInverted'),
                borderColor: get('semanticColors.button.primary.borderHoverInverted')
            },
            '&:disabled': {
                color: get('semanticColors.button.primary.textDisabledInverted'),
                fill: get('semanticColors.button.primary.iconDisabledInverted'),
                background: get('semanticColors.button.primary.backgroundDisabledInverted'),
                borderColor: get('semanticColors.button.primary.borderDisabledInverted')
            }
        },
        secondary: {
            color: get('semanticColors.button.secondary.textInverted'),
            fill: get('semanticColors.button.secondary.iconInverted'),
            background: get('semanticColors.button.secondary.backgroundInverted'),
            borderColor: get('semanticColors.button.secondary.borderInverted'),

            '&:hover': {
                color: get('semanticColors.button.secondary.textHoverInverted'),
                fill: get('semanticColors.button.secondary.iconHoverInverted'),
                background: get('semanticColors.button.secondary.backgroundHoverInverted'),
                borderColor: get('semanticColors.button.secondary.borderHoverInverted')
            },
            '&:disabled': {
                color: get('semanticColors.button.secondary.textDisabledInverted'),
                fill: get('semanticColors.button.secondary.iconDisabledInverted'),
                background: get('semanticColors.button.secondary.backgroundDisabledInverted'),
                borderColor: get('semanticColors.button.secondary.borderDisabledInverted')
            }
        },
        danger: {
            color: get('semanticColors.button.danger.textInverted'),
            fill: get('semanticColors.button.danger.iconInverted'),
            background: get('semanticColors.button.danger.backgroundInverted'),
            borderColor: get('semanticColors.button.danger.borderInverted'),

            '&:hover': {
                color: get('semanticColors.button.danger.textHoverInverted'),
                fill: get('semanticColors.button.danger.iconHoverInverted'),
                background: get('semanticColors.button.danger.backgroundHoverInverted'),
                borderColor: get('semanticColors.button.danger.borderHoverInverted')
            },
            '&:disabled': {
                color: get('semanticColors.button.danger.textDisabledInverted'),
                fill: get('semanticColors.button.danger.iconDisabledInverted'),
                background: get('semanticColors.button.danger.backgroundDisabledInverted'),
                borderColor: get('semanticColors.button.danger.borderDisabledInverted')
            }
        }
    }
});

const Button: React.FC<ButtonProps> = styled(BaseButton).attrs({ theme })<ButtonProps>`
    transition: background ease 200ms, border-color ease 200ms, color ease 200ms, fill ease 200ms;

    ${p => (p.inverted ? invertedVariantStyles : variantStyles)};
`;

Button.defaultProps = {
    size: 'medium',
    variant: 'primary'
};

export { Button, ButtonProps };
