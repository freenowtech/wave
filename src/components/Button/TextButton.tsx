import * as React from 'react';
import styled from 'styled-components';
import { ResponsiveValue, variant } from 'styled-system';
import { get } from '../../utils/themeGet';

import { BaseButton, BaseButtonProps } from './BaseButton';

interface TextButtonProps extends BaseButtonProps {
    /**
     * adjust color for display on a dark background
     */
    inverted?: boolean;
    /**
     * Define base colors
     */
    variant?: ResponsiveValue<'default' | 'danger'>;
}

const variantStyles = variant({
    variants: {
        default: {
            color: get('semanticColors.textButton.primary.text'),
            fill: get('semanticColors.textButton.primary.icon'),

            '&:hover': {
                color: get('semanticColors.textButton.primary.textHover'),
                fill: get('semanticColors.textButton.primary.iconHover')
            },

            '&:disabled': {
                color: get('semanticColors.textButton.primary.textDisabled'),
                fill: get('semanticColors.textButton.primary.iconDisabled')
            }
        },
        danger: {
            color: get('semanticColors.textButton.danger.text'),
            fill: get('semanticColors.textButton.danger.icon'),

            '&:hover': {
                color: get('semanticColors.textButton.danger.textHover'),
                fill: get('semanticColors.textButton.danger.iconHover')
            },

            '&:disabled': {
                color: get('semanticColors.textButton.danger.textDisabled'),
                fill: get('semanticColors.textButton.danger.iconDisabled')
            }
        }
    }
});

const invertedVariantStyles = variant({
    variants: {
        default: {
            color: get('semanticColors.textButton.primary.textInverted'),
            fill: get('semanticColors.textButton.primary.iconInverted'),

            '&:hover': {
                color: get('semanticColors.textButton.primary.textHoverInverted'),
                fill: get('semanticColors.textButton.primary.iconHoverInverted')
            },

            '&:disabled': {
                color: get('semanticColors.textButton.primary.textDisabledInverted'),
                fill: get('semanticColors.textButton.primary.iconDisabledInverted')
            }
        },
        danger: {
            color: get('semanticColors.textButton.danger.textInverted'),
            fill: get('semanticColors.textButton.danger.iconInverted'),

            '&:hover': {
                color: get('semanticColors.textButton.danger.textHoverInverted'),
                fill: get('semanticColors.textButton.danger.iconHoverInverted')
            },

            '&:disabled': {
                color: get('semanticColors.textButton.danger.textDisabledInverted'),
                fill: get('semanticColors.textButton.danger.iconDisabledInverted')
            }
        }
    }
});

const TextButton: React.FC<TextButtonProps> = styled(BaseButton)<TextButtonProps>`
    transition: color 200ms, fill 200ms;

    ${props => (props.inverted ? invertedVariantStyles(props) : variantStyles(props))};
`;

TextButton.defaultProps = {
    size: 'medium',
    variant: 'default'
};

export { TextButton, TextButtonProps };
