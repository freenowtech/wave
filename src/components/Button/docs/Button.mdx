---
name: Button
menu: Components
route: /components/button
---

import { Playground } from 'docz';
import { Button } from '../Button';
import { TextButton } from '../TextButton';
import { Box } from '../../Box/Box';
import { Headline } from '../../Headline/Headline';
import { Text } from '../../Text/Text';
import { TrashBinIcon, CloseIcon } from '../../../icons/index.ts';
import { ButtonPropsTable } from './ButtonPropsTable.tsx';
import { Combination } from '../../../docs/Combination'
import { StyledSystemLinks } from '../../../docs/StyledSystemLinks'

# Button

Renders a `button` tag. This module exposes two types of buttons `<TextButton />` and `<Button />`.
Take a look to the [MDN web docs - Button](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) to know more
about some of the default props supported by a button along with some accessibility concerns.

The purpose of the buttons is to trigger an interaction like order, confirmation or submit. For navigation
targets consider a link. Use the primary button only once per screen and avoid putting two primary buttons next to each other.
Use the secondary button for less important actions and in multi button scenarios.

## Properties

<ButtonPropsTable />
<br/>

<StyledSystemLinks component="Button" supportedProps={ ['margin', 'width'] }/>
<StyledSystemLinks component="TextButton" supportedProps={ ['margin', 'width'] }/>

## Examples

<Playground>
    <Button>Regular</Button>
    <TextButton>Text Only</TextButton>
</Playground>

### Icons

Sometimes an icon can help clarify the usage of the button. The recommended sizes for the icons:
* `<IconName size={20} />` — 20px for regular button
* `<IconName size={18} />` — 18px for small button

<Playground>
    <Button variant="danger">
        <TrashBinIcon size={20} /> Remove
    </Button>

    <Button size='small' ml={2}>
      <CloseIcon size={18} /> Remove
    </Button>
</Playground>

## Combinations

### Button

<Combination disabled={[false, true]} variant={["primary", "secondary", "danger"]} size={["medium", "small"]} inverted={[false, true]}>
    {(props, i) => <Button key={i} {...props}>Button</Button>}
</Combination>

### TextButton

<Combination disabled={[false, true]} variant={["default", "danger"]} size={["medium", "small"]} inverted={[false, true]}>
    {(props, i) => <TextButton key={i} {...props}>Button</TextButton>}
</Combination>
