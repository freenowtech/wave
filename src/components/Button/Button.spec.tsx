import React from 'react';
import { render } from '@testing-library/react';
import { Button } from './Button';
import { SemanticColors } from '../../essentials';

describe('standard button', () => {
    it('renders by default the primary variant', () => {
        const { container: primaryTree } = render(<Button variant="primary">Click</Button>);
        const { container: defaultTree } = render(<Button>Click</Button>);

        expect(primaryTree).toEqual(defaultTree);
    });

    describe('primary', () => {
        it('renders as design system suggests', () => {
            const { container } = render(<Button variant="primary">Click</Button>);
            expect(container.firstChild).toHaveStyle(`
              background: ${SemanticColors.button.primary.background};
              color: ${SemanticColors.button.primary.text};
            `);
            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe('secondary', () => {
        it('renders as design system suggests', () => {
            const { container } = render(<Button variant="secondary">Click</Button>);
            expect(container.firstChild).toHaveStyle(`
              background: ${SemanticColors.button.secondary.background};
              color: ${SemanticColors.button.secondary.text};
              border-color: ${SemanticColors.button.secondary.border};
            `);
            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe('danger', () => {
        it('renders as design system suggests', () => {
            const { container } = render(<Button variant="danger">Click</Button>);
            expect(container.firstChild).toHaveStyle(`
              background: ${SemanticColors.button.danger.background};
              color: ${SemanticColors.button.danger.text};
            `);
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});

describe('inverted button', () => {
    it('renders by default the primary variant', () => {
        const { container: primaryTree } = render(
            <Button inverted variant="primary">
                Click
            </Button>
        );
        const { container: defaultTree } = render(<Button inverted>Click</Button>);

        expect(primaryTree).toEqual(defaultTree);
    });

    describe('primary', () => {
        it('renders as design system suggests', () => {
            const { container } = render(
                <Button variant="primary" inverted>
                    Click
                </Button>
            );
            expect(container.firstChild).toHaveStyle(`
              background: ${SemanticColors.button.primary.backgroundInverted};
              color: ${SemanticColors.button.primary.textInverted};
            `);
            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe('secondary', () => {
        it('renders as design system suggests', () => {
            const { container } = render(
                <Button variant="secondary" inverted>
                    Click
                </Button>
            );

            expect(container.firstChild).toHaveStyle(`
              background: transparent;
              color: ${SemanticColors.button.secondary.textInverted};
              border-color: ${SemanticColors.button.secondary.borderInverted};
            `);
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
