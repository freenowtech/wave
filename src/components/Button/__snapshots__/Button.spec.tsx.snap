// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inverted button primary renders as design system suggests 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #096BDB;
  fill: #096BDB;
  background: #FFFFFF;
  border-color: #FFFFFF;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 svg path {
  fill: inherit;
}

.c0:hover {
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #096BDB;
  border-color: #096BDB;
}

.c0:disabled {
  color: #9CA7B4;
  fill: #9CA7B4;
  background: #637689;
  border-color: #637689;
}

<button
  class="sc-AxjAm c0"
>
  Click
</button>
`;

exports[`inverted button secondary renders as design system suggests 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #FFFFFF;
  fill: #FFFFFF;
  background: transparent;
  border-color: #FFFFFF;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 svg path {
  fill: inherit;
}

.c0:hover {
  color: #001E3E;
  fill: #001E3E;
  background: #FFFFFF;
  border-color: #FFFFFF;
}

.c0:disabled {
  color: #637689;
  fill: #637689;
  background: transparent;
  border-color: #637689;
}

<button
  class="sc-AxjAm c0"
>
  Click
</button>
`;

exports[`standard button danger renders as design system suggests 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #FF5E05;
  border-color: #FF5E05;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 svg path {
  fill: inherit;
}

.c0:hover {
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #BF4704;
  border-color: #BF4704;
}

.c0:disabled {
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #C6CDD4;
  border-color: #C6CDD4;
}

<button
  class="sc-AxjAm c0"
>
  Click
</button>
`;

exports[`standard button primary renders as design system suggests 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #001E3E;
  border-color: #001E3E;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 svg path {
  fill: inherit;
}

.c0:hover {
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #000F1F;
  border-color: #000F1F;
}

.c0:disabled {
  color: #FFFFFF;
  fill: #FFFFFF;
  background: #C6CDD4;
  border-color: #C6CDD4;
}

<button
  class="sc-AxjAm c0"
>
  Click
</button>
`;

exports[`standard button secondary renders as design system suggests 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #001E3E;
  fill: #001E3E;
  background: #FFFFFF;
  border-color: #C6CDD4;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 svg path {
  fill: inherit;
}

.c0:hover {
  color: #001E3E;
  fill: #001E3E;
  background: #F1F2F4;
  border-color: #C6CDD4;
}

.c0:disabled {
  color: #C6CDD4;
  fill: #C6CDD4;
  background: #FFFFFF;
  border-color: #C6CDD4;
}

<button
  class="sc-AxjAm c0"
>
  Click
</button>
`;
